### DISCUSSION TITLE
Input.http buffer gradually fills

### DISCUSSION THREAD

**Discussion by maardsma** [savonet/liquidsoap#1897] [Q&A] [answered]
I am using input.http to download two streams, add them together, compress, and stream back out. My sources dynamic and I am using an input.harbor internally so I can shutdown and change the dynamic sources internally.

When I look at the buffer debug file output, I get two numbers like this.

1.468109 51570
1.475545 52276
1.502378 40192
1.523011 32872
1.549844 29520
1.570446 30931
1.570459 34900
1.590917 32342
1.617656 29785
1.637956 31637
1.658411 33048
1.685195 30490
1.705816 31901
1.726080 33312
1.752877 30755
1.773165 32607
1.793547 34018
1.820732 31019
1.820775 45308
1.841215 32871
1.867977 29873

What do these mean? Is the first one buffer contents in seconds duration?

Mark

**Comment by toots** [comment]
Hi Mark,

Yes, this the buffer content duration. Are you getting the data from a real-time source such as a icecast source or from a regular file served over http?

In the latter case, liquidsoap will download as fast as it can, leading to be buffer being filled up as you are seeing. In this case, you should set the `self_sync` parameter to `false`:
```ruby
s = input.http(self_sync=false, ...)
```

**Reply by maardsma** [reply]
The data is from another icecast server.

Looks like self_sync is not in version 1.4.4. But probably not relevant since I am downloading from a realtime source.

I installed with OPAM without specifying a version. How do I upgrade my install to the dev version?

Mark

**Reply by toots** [reply]
Ha I see. `liquidsoap.2.0.0~rc1` is the default in `opam` now if you run `opam update`.

That being said, `self_sync` will not solve your issue. I'd recommend to try `2.0.0`, though, it is our target for debugging currently and, also, the implementation for `input.http` was totally redone and now uses `ffmpeg` under the hood (make sure to install it as well!). This might have fixed your issue.

**Reply by maardsma** [reply]
I upgraded to 2.0 and I attempted to install ffmpeg via "opam install ffmpeg" and I get this error during install:

What is the appropriate place to ask for help with this?
----

```
[ERROR] The compilation of ffmpeg-avutil.1.0.0~rc1 failed at "dune build -p ffmpeg-avutil -j 1 @install".

#=== ERROR while compiling ffmpeg-avutil.1.0.0~rc1 ============================#
# context     2.1.0~rc2 | linux/x86_64 | ocaml-base-compiler.4.08.0 | https://opam.ocaml.org#81d2d054
# path        ~/.opam/4.08.0/.opam-switch/build/ffmpeg-avutil.1.0.0~rc1
# command     ~/.opam/opam-init/hooks/sandbox.sh build dune build -p ffmpeg-avutil -j 1 @install
# exit-code   1
# env-file    ~/.opam/log/ffmpeg-avutil-29035-5aaa46.env
# output-file ~/.opam/log/ffmpeg-avutil-29035-5aaa46.out
### output ###
# avutil_stubs.c:1641:13: error: request for member ‘sw_format’ in something not a structure or union
# [...]
#              ^~
# avutil_stubs.c:1642:13: error: request for member ‘width’ in something not a structure or union
#    frames_ctx->width = Int_val(_width);
#              ^~
# avutil_stubs.c:1643:13: error: request for member ‘height’ in something not a structure or union
#    frames_ctx->height = Int_val(_height);
#              ^~
# avutil_stubs.c:1646:9: warning: implicit declaration of function ‘av_hwframe_ctx_init’; did you mean ‘av_frame_clone’? [-Wimplicit-function-declaration]
#    ret = av_hwframe_ctx_init(hw_frames_ref);
#          ^~~~~~~~~~~~~~~~~~~
#          av_frame_clone
```

**Reply by toots** [reply]
It looks like you have an old version of the ffmpeg libraries. What OS are using?

**Reply by maardsma** [reply]
Ubuntu 18.4

**Comment by maardsma** [comment, marked as answer]
Issue was resolved by using Ubuntu Server 20.04 OS instead of Ubuntu Server 18.4 OS.

**Reply by toots** [reply]
Great to hear, thanks!
